src/navmazing/__init__.py

```diff
@@ -0,0 +1,43 @@
+"""A simplified navigation framework with prerequisite, object and intelligence support.
+
+An example is below::
+
+    from navmazing import Navigate, NavigateStep, NavigateToSibling
+
+    navigate = Navigate()
+
+    class Provider(object):
+        def __init__(self, name):
+            self.name = name
+
+
+    @navigate.register(Provider, 'New')
+    class AddANewProvider(NavigateStep)
+        prerequisite = NavigateToSibling('All')
+
+        def step(self):
+            click('Add New Button')
+
+    @navigate.register(Provider, 'All')
+    class ShowAllProviders(NavigateStep)
+        def am_i_here(self):
+            return check_if_i_am_already_on_page()
+
+        def step(self):
+            click('All button')
+
+"""
+from __future__ import annotations
+from ._errors import NavigationTriesExceeded
```

Comment:
What is the advantage of using `_` as prefix if module names? Or is it just for convention?

Reply:
That's the convention for private modules

Other parties ought not to import from those